### PR TITLE
feat: add 3 additional paycor streams

### PR DESCRIPTION
**Pull request by themightychris**
We need to add 3 new streams to the Paycor tap:

- [ ] `time_off_requests`
    - Endpoint: `/v1/legalentities/${LEGAL_ENTITY_ID}/timeoffrequests`
    - See example call in `api-sandbox/get-time-off-requests`
    - Start and end date are required even though docs say they aren't
    - Start date must be set to 1 year ago + 1 day
    - End date must be start date + 90 days (actually needs calendar math or call will fail)
    - Within each date range, results may be paged the same way as the employees API
    - If the requested date range has no results, it returns 400—ignore and move on to the next date range
    - Crawl 1 years into the future
- [ ] `employee_custom_fields`
    - Endpoint: `/v1/employees/${EMPLOYEE_ID}/customfields`
    - See example call in `api-sandbox/get-employee-custom-fields`
    - Must be called for each employee, do within loop when processing "employees" stream (it's ok to interlace records from different streams in outpu)
    - Within each employee, results may be paged the same way as the employees API
- [ ] `persons`
    - Endpoint: `/v1/legalEntities/122743/persons?include=All`
    - See example call in `api-sandbox/get-persons`
    - Must be called for each employee, do within loop when processing "employees" stream (it's ok to interlace records from different streams in outpu)
    - Within each employee, results may be paged the same way as the employees API

For each added stream, JSON Schema must be built from Paycor's busted OpenAPI schema:

- Download OpenAPI spec via button at top of <https://developers.paycor.com/explore>
- Find `"EmployeeTimeOffRequest"`, `"EmployeeCustomField"`, and `"Person"` definitions
- Replicate property list
- Nested references must be flattened—in VScode you can ctrl+click the refs to jump to them
- Make field names match case in actual responses
- Adapt to match shape of JSONSchema, use employees.json as example
- Test with validator documented in main README

### CODE REVIEW

**Comment by casscauley** [comment]
I automated correcting the swagger schema using the `create_schema.py` script. The sync seems to be running fine, but I'm not completely sure everything is right.
* Running `tap-paycor --config tap_config.json --catalog catalog.json` seemed to output everything fine
* I tried modifying the schema to make it break (adding additional required fields, changing the type on the fields) and it didn't seem to affect anything.

**Comment by themightychris** [comment]
@chriscauley I've completed all the schema and updated the catalog

File: tap_paycor/__init__.py [hunk 1]
```diff
@@ -77,56 +90,92 @@ def refresh_token(args):
...
+    has_more_results = True
+    ids = []
+
+    while True:
+        r = requests.get(url, headers=headers)
+        tap_data = r.json()
+        if 'is invalid or has no' in tap_data.get('Detail', ''):
+            # When there are no entities, Paycor sends back a 400 with a message like
+            # 'Either Legal Entity ID ### is invalid or has no TimeOff requests.'
+            break
```
Comment (casscauley) [inline tap_paycor/__init__.py:181]: The above is how we're handling the "empty list is a 400 error" quirk

File: tap_paycor/__init__.py [hunk 2]
```diff
@@ -77,56 +90,92 @@ def refresh_token(args):
...
+            schema=stream.schema.to_dict(),
+            key_properties=stream.key_properties,
+        )
+        if stream.tap_stream_id == 'employee_custom_fields':
+            if employee_ids is None:
+                # this must be handled in employees
+                raise ValueError('You must tap employees before employee_custom_fields')
+            for employee_id in employee_ids:
+                get_records(config, stream, employee_id)
+            continue
```
Comment (casscauley) [inline tap_paycor/__init__.py]: I believe this is the only way that I deviated from the @themightychris' instructions. Instead I:

* Do employees first (changes in catalog.json) and then set aside the `employee_ids` (line 120-121 below)
* When processing the `employee_custom_fields`, verify that employee_ids was set (not `None`) and then loop over the `employee_ids`

**Pull request merged**